test/e2e/scenario/recorder.scenario.ts

```diff
@@ -654,17 +694,305 @@ describe('recorder', () => {
         expect(segment.records.slice(3).every((record) => record.type !== RecordType.FullSnapshot)).toBe(true)
       })
   })
+
+  describe('layout viewport properties - ', () => {
+    createTest('getWindowWidth/Height reports using layout viewport dimensions')
+      .withRum({ enableExperimentalFeatures: ['visualviewport'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(html`${VIEWPORT_META_TAGS}`)
+      .run(async ({ events }) => {
+        /**
+         * InnerWidth/Height on some devices/browsers are changed by pinch zoom
+         * We need to ensure that our measurements are not affected by pinch zoom
+         */
+        if (isGestureUnsupported()) {
+          return // No Fallback test
+        }
+        await resetViewport()
+
+        const { innerWidth, innerHeight } = (await browserExecute(() => ({
+          innerWidth: window.innerWidth,
+          innerHeight: window.innerHeight,
+        }))) as { innerWidth: number; innerHeight: number }
+
+        const initialVisualViewport = await getVisualViewport()
+        await pinchZoom(150)
+        await pinchZoom(150)
+        await sleep(210)
+        const nextVisualViewport = await getVisualViewport()
+
+        await browserExecute(() => {
+          window.dispatchEvent(new Event('resize'))
+        })
+        await sleep(210)
+
+        await flushEvents()
+        const segment = getLastSegment(events)
+        const ViewportResizeRecords = findAllIncrementalSnapshots(segment, IncrementalSource.ViewportResize)
+
+        const lastViewportResizeRecord = ViewportResizeRecords.slice(-1)[0].data as ViewportResizeData
```

Comment:
no clue why:

- 150
- two pinches
- need to sleep
- 210
- window resize

it could worth some explanation

Reply:
Great questions:
- first off, not sure why sleeping helps, but it reliably helps with flakiness. 210ms was picked and used consistently because it was the highest amount that the SDK throttles that also feels low enough (100-200ms)- though FMU throttling should not be responsible for flakiness, but I have no other leads.
- Pinch zooms are not setting to 1.5x zoom, but rather moving fingers 150px in each direction- which can exceed the bounds of the device's screen causing an error- so two pinches are used to zoom in "a bunch" (an unknown/undefined amount). Having less pinch zoom (1 call) has less clear differences from `innerWidth/height`, while having 3 would be excessive IMO.
- The dispatched resize event triggers the SDK to take a record which is what we are testing.

The purpose of this test is to check that our new browser dimensions are measuring the layout viewport, and so they do not change when pinch zoom is applied.  I'll add a comment in the code.

Reply:
> first off, not sure why sleeping helps, but it reliably helps with flakiness. 210ms was picked and used consistently because it was the highest amount that the SDK throttles that also feels low enough (100-200ms)- though FMU throttling should not be responsible for flakiness, but I have no other leads.

From past experiences, my take is that relying on timeout is rarely a persistent solution. Could we find why we need to wait and see if we can rely on something else than time?

> Pinch zooms are not setting to 1.5x zoom, but rather moving fingers 150px in each direction- which can exceed the bounds of the device's screen causing an error- so two pinches are used to zoom in "a bunch" (an unknown/undefined amount). Having less pinch zoom (1 call) has less clear differences from innerWidth/height, while having 3 would be excessive IMO.

cf https://github.com/DataDog/browser-sdk/pull/1118#discussion_r733431423, it could be nice to abstract this operation and add this kind of explanation with it can also help